### ISSUE TITLE
Doesn't build on OS X

### ISSUE THREAD

**Issue by killercup** [swatteau/sokoban-rs#1]
I wanted to try this out, but `cargo build --release` fails. Is there anything special I maybe need to do? Does this maybe not work with clang?

The error seems to be this one:

``` text
error: linking with `cc` failed: exit code: 1
```

which leads to this

``` text
ld: library not found for -lSDL2_image
```

The full output is this:

``` text
$ cargo build --release --verbose
       Fresh bitflags v0.2.1
       Fresh bitflags v0.3.2
       Fresh libc v0.1.10
       Fresh rustc-serialize v0.3.16
       Fresh winapi v0.2.2
       Fresh bitflags v0.1.1
       Fresh winapi-build v0.1.1
       Fresh xml-rs v0.1.26
       Fresh advapi32-sys v0.1.2
       Fresh rand v0.3.10
       Fresh num v0.1.27
       Fresh sdl2-sys v0.6.2
       Fresh sdl2 v0.8.0
   Compiling sokoban-rs v0.1.1 (file:///Users/pascal/Projekte/tools/sokoban-rs)
     Running `rustc src/main.rs --crate-name sokoban_rs --crate-type bin -C opt-level=3 --out-dir /Users/pascal/Projekte/tools/sokoban-rs/target/release --emit=dep-info,link -L dependency=/Users/pascal/Projekte/tools/sokoban-rs/target/release -L dependency=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps --extern bitflags=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libbitflags-c495827ddcd6a202.rlib --extern xml=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libxml-ec337cc3dc06d297.rlib --extern sdl2_image=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libsdl2_image-91ae984502e83bb5.rlib --extern sdl2=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libsdl2-b3ed89a7752e0295.rlib`
       Fresh sdl2_image v0.2.5 (https://github.com/xsleonard/rust-sdl2_image#33381b39)
error: linking with `cc` failed: exit code: 1
note: "cc" "-m64" "-L" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/sokoban_rs.0.o" "-o" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/sokoban_rs" "-Wl,-dead_strip" "-nodefaultlibs" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libxml-ec337cc3dc06d297.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libsdl2_image-91ae984502e83bb5.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libsdl2-b3ed89a7752e0295.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libbitflags-425420435a0b4460.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libbitflags-c495827ddcd6a202.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libbitflags-dd68b8369bcd8ff0.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libsdl2_sys-c2fbfeba65e652ce.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libnum-397f282c1d72fe58.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/librand-aeb0317fb0b04603.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/liblibc-144c435538abd757.rlib" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/librustc_serialize-7ff5bfc027146194.rlib" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib/libstd-35017696.rlib" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib/libcollections-35017696.rlib" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib/librustc_unicode-35017696.rlib" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib/librand-35017696.rlib" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib/liballoc-35017696.rlib" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib/liballoc_jemalloc-35017696.rlib" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib/liblibc-35017696.rlib" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib/libcore-35017696.rlib" "-L" "/Users/pascal/Projekte/tools/sokoban-rs/target/release" "-L" "/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps" "-L" "/Users/pascal/.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib" "-L" "/Users/pascal/Projekte/tools/sokoban-rs/.rust/lib/x86_64-apple-darwin" "-L" "/Users/pascal/Projekte/tools/sokoban-rs/lib/x86_64-apple-darwin" "-l" "SDL2_image" "-l" "SDL2" "-l" "SDL2" "-l" "c" "-l" "m" "-l" "System" "-l" "pthread" "-l" "c" "-l" "m" "-Wl,-rpath,@loader_path/../../../../../.multirust/toolchains/nightly/lib/rustlib/x86_64-apple-darwin/lib" "-Wl,-rpath,/usr/local/lib/rustlib/x86_64-apple-darwin/lib" "-l" "compiler-rt"
note: ld: warning: directory not found for option '-L/Users/pascal/Projekte/tools/sokoban-rs/.rust/lib/x86_64-apple-darwin'
ld: warning: directory not found for option '-L/Users/pascal/Projekte/tools/sokoban-rs/lib/x86_64-apple-darwin'
ld: library not found for -lSDL2_image
clang: error: linker command failed with exit code 1 (use -v to see invocation)

error: aborting due to previous error
Could not compile `sokoban-rs`.

Caused by:
  Process didn't exit successfully: `rustc src/main.rs --crate-name sokoban_rs --crate-type bin -C opt-level=3 --out-dir /Users/pascal/Projekte/tools/sokoban-rs/target/release --emit=dep-info,link -L dependency=/Users/pascal/Projekte/tools/sokoban-rs/target/release -L dependency=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps --extern bitflags=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libbitflags-c495827ddcd6a202.rlib --extern xml=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libxml-ec337cc3dc06d297.rlib --extern sdl2_image=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libsdl2_image-91ae984502e83bb5.rlib --extern sdl2=/Users/pascal/Projekte/tools/sokoban-rs/target/release/deps/libsdl2-b3ed89a7752e0295.rlib` (exit code: 101)
```

---

I'm using this rust which I installed with multirust:

``` text
$ rustc -vV
rustc 1.4.0-nightly (8f1b0aa32 2015-08-21)
binary: rustc
commit-hash: 8f1b0aa32552f2e694aa8702ff2cd6d9a0e894f1
commit-date: 2015-08-21
host: x86_64-apple-darwin
release: 1.4.0-nightly
```

``` text
$ clang --version
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.4.0
Thread model: posix
```

``` text
$ ld -v
@(#)PROGRAM:ld  PROJECT:ld64-242.2
configured to support archs: armv6 armv7 armv7s arm64 i386 x86_64 x86_64h armv6m armv7m armv7em
LTO support using: LLVM version 3.6.0svn
```

**Comment by swatteau** [comment]
I think you need to install the development files for SDL2 and SDL2_image like you'd do for a C project. On Linux, these files come with the libsdl2-dev and libsdl2-image-dev packages. I don't know how they are called on OSX.

**Comment by swatteau** [comment]
I just added some precisions in the README file about the requirements for building sokoban-rs. Please let me know if that fixes your problem.
Thanks for your feedback.

**Comment by killercup** [comment]
Thanks for your comments!

**tl;dr** You might want to add `brew install sdl2 sdl2_image` to the install instructions for OS X :smile: 

I installed SDL2 with `brew install sdl2` (as suggested by the _rust-sdl2_ Readme), but still got the same error (`ld: library not found for -lSDL2_image`). Luckily, I noticed that there's a _sdl2_image_ package as well, so after another `brew install sdl2_image` everything seems to work! (I ended up using `env LIBRARY_PATH="$LIBRARY_PATH:/usr/local/lib" cargo build --release` but I'm not sure the extra env variable is necessary).
